lib/active_merchant/billing/gateways/jetpay_v2.rb

```diff
@@ -0,0 +1,406 @@
+        "962" =>  "Invalid PIN key (Unknown KSN).",
+        "981" =>  "Invalid AVS.",
+        "987" =>  "Issuer Unavailable.",
+        "988" =>  "System error SD.",
+        "989" =>  "Database Error.",
+        "992" =>  "Transaction Timeout.",
+        "996" =>  "Bad Terminal ID.",
+        "997" =>  "Message rejected by association.",
+        "999" =>  "Communication failure",
+        nil   =>  "No response returned (missing credentials?)."
+      }
+
+      def initialize(options = {})
+        requires!(options, :login)
+        super
+      end
+
+      def purchase(money, credit_card, options = {})
+        commit(money, build_sale_request(money, credit_card, options))
+      end
+
+      def authorize(money, credit_card, options = {})
+        commit(money, build_authonly_request(money, credit_card, options))
+      end
+
+      def capture(money, reference, options = {})
+        split_authorization = reference.split(";")
+        transaction_id = split_authorization[0]
+        token = split_authorization[3]
+        commit(money, build_capture_request(transaction_id, money, options), token)
+      end
+
+      def void(reference, options = {})
+        transaction_id, approval, amount, token = reference.split(";")
+        commit(amount.to_i, build_void_request(amount.to_i, transaction_id, approval, token, options), token)
+      end
+
+      def credit(money, credit_card, options = {})
+        commit(money, build_credit_request('CREDIT', money, nil, credit_card, nil, options))
+      end
+
+      def refund(money, reference, options = {})
+        split_authorization = reference.split(";")
+        transaction_id = split_authorization[0]
+        token = split_authorization[3]
+        commit(money, build_credit_request('CREDIT', money, transaction_id, nil, token, options))
+      end
+
+      def verify(credit_card, options={})
+        authorize(0, credit_card, options)
+      end
+
+      def supports_scrubbing
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((>)\d+(</CardNum>)), '\1[FILTERED]\2').
+          gsub(%r((<CVV2>)\d+(</CVV2>)), '\1[FILTERED]\2')
+      end
+
+      private
+
+      def build_xml_request(transaction_type, options = {}, transaction_id = nil, &block)
+        xml = Builder::XmlMarkup.new
+        xml.tag! 'JetPay', 'Version' => API_VERSION do
+          # Basic values needed for any request
+          xml.tag! 'TerminalID', @options[:login]
+          xml.tag! 'TransactionType', transaction_type
+          xml.tag! 'TransactionID', transaction_id.nil? ? generate_unique_id.slice(0, 18) : transaction_id
+          xml.tag! 'Origin', options[:origin] || 'INTERNET'
+          xml.tag! 'IndustryInfo', 'Type' => options[:industry_info] || 'ECOMMERCE'
+          xml.tag! 'Application', options[:application] || 'n/a', 'Version' => options[:application_version] || '1.0'
```

Comment:
I think this syntax will error out? ` 'n/a', 'Version' => options[:application_version]` both the comma after `n/a` and the `'Version' => options[...]`. 

I guess that makes me wonder if we can just simplify it with:

`options[:application] || '1.0`

Unless I'm missing something about the JetPay v2 API where we might need to send something related to that middle value.